NitroxClient/MonoBehaviours/Gui/InGame/Modal.cs

```diff
@@ -0,0 +1,170 @@
+            IngameMenu.main.OnDeselect();
+            IsAvoidableBypass = false;
+        }
+        else
+        {
+            IngameMenu.main.OnDeselect();
+        }
+    }
+
+    /// <summary>
+    /// This creates the modal when showing it for the first time, you can't modify it afterwards
+    /// </summary>
+    private void InitSubWindow()
+    {
+        if (!IngameMenu.main)
+        {
+            throw new NotSupportedException($"Cannot show ingame subwindow {SubWindowName} because the ingame window does not exist.");
+        }
+
+        if (!modalSubWindow)
+        {
+            GameObject derivedSubWindow = IngameMenu.main.transform.Find("QuitConfirmation").gameObject;
+            modalSubWindow = Instantiate(derivedSubWindow, IngameMenu.main.transform, false);
+            modalSubWindow.name = SubWindowName;
+
+            // Styling.
+            RectTransform main = modalSubWindow.GetComponent<RectTransform>();
+            main.sizeDelta = new Vector2(700, 195);
+
+
+            RectTransform messageTransform = modalSubWindow.FindChild("Header").GetComponent<RectTransform>();
+            messageTransform.sizeDelta = new Vector2(700, 195);
+         }
+
+        // Will happen either it's initialized or not
+        UpdateModal();
+    }
+
+    /// <summary>
+    /// Update the modal with informations that may change from one Show() to another
+    /// </summary>
+    private void UpdateModal()
+    {
+        text = modalSubWindow.FindChild("Header").GetComponent<Text>();
+        text.text = ModalText;
+
+        GameObject buttonYesObject = modalSubWindow.FindChild("ButtonYes");
+        GameObject buttonNoObject = modalSubWindow.FindChild("ButtonNo");
+        Button yesButton = buttonYesObject.GetComponent<Button>();
+
+        // We need to reinitialize onClick to avoid keeping Persisted Events (which are set manually inside Unity's Editor)
+        yesButton.onClick = new Button.ButtonClickedEvent();
+        yesButton.onClick.AddListener(() => { ClickYes(); });
+        buttonYesObject.GetComponentInChildren<Text>().text = YesButtonText;
+
+        if (HideNoButton && buttonNoObject)
+        {
+            DestroyImmediate(buttonNoObject);
+            buttonYesObject.transform.position = new Vector3(modalSubWindow.transform.position.x / 2, buttonYesObject.transform.position.y, buttonYesObject.transform.position.z); // Center Button
+            return;
+        }
+
+        Button noButton = buttonNoObject.GetComponent<Button>();
+        noButton.onClick = new Button.ButtonClickedEvent();
+        noButton.onClick.AddListener(() => { ClickNo(); });
+        buttonNoObject.GetComponentInChildren<Text>().text = NoButtonText;
+    }
+
+    public abstract void ClickYes();
+    public abstract void ClickNo();
+
+    private IEnumerator Show_Impl()
+    {
+        // Execute frame-by-frame to allow UI scripts to initialize.
+        InitSubWindow();
+        yield return null;
```

Comment:
`yield break;` or if this isn't working `yield return new WaitForEndOfFrame();`

Reply:
`yield break;` makes following code unreachable apparently